### ISSUE TITLE
Implement filter type interface that fires a callback on ANY augur contract tx

### ISSUE THREAD

**Issue by scottzer0** [AugurProject/augur.js#2]
It watches for new transactions and fire a callback and returns a nice data structure of the tx that is augur related (ie. share purchased/sold, etc.)

for starters, it would be nice to know when a new market/event has hit the chain so i can update the client. (don't want to have to poll all markets, all the time!)

**Comment by scottzer0** [comment]
I'm now thinking this is just:
`Augur.onTX(txHash, callback)`

**Comment by tinybike** [comment]
Implemented in 07adf271e91eebdf5c56acbf23c7999aa7b298c3.
```javascript
// create filter for transactions on any of the augur contract addresses
augur.filters.start_contracts_listener();

// check filter for new transactions every 5 seconds
augur.filters.heartbeat(function (message) {
    console.log("new transaction:", message);
});

// stop heartbeat polling
augur.filters.stop_heartbeat();

// stop filtering for contract transactions
augur.filters.clear_contracts_filter();
```
**Important note:** due to a geth bug for multi-address filters, the all-augur-contracts filters **will not work** until ethereum/go-ethereum#1751 is merged.  I've merged this locally and verified that the filters work properly afterwards.
